### ISSUE TITLE
tower: path forward for `tracing-tower`

### ISSUE THREAD

**Issue by hawkw** [tokio-rs/tracing#238]
### Crates

- `tracing-tower`
- `tracing-tower-http`

### Motivation
There is demand for a `0.0.1-alpha.2` release of `tracing-subscriber`. However, this crate now depends on `tracing-tower` (for the load generator example). `tracing-tower` is currently unpublished. Before we publish `tracing-tower`, there are a couple of questions that need to be worked out.

### Questions 

1. Should this crate even live in `tracing`'s github repo? Since it exists to provide implementations of `tower` traits, perhaps it should be renamed to `tower-tracing` and live in the `tower` github repo instead. Input from `tower` folks (@davidbarsky, @LucioFranco, @jonhoo) would be great...
2. The `tracing-tower-http` crate provides batteries-included functionality for instrumenting services whose request and response types are `http::Request`/`http::Response`. However, it can now implemented in `tracing-tower` with a feature flag for the `http` dependency. IMO, this is better, as it reduces the number of very small crates in `tracing`. Should we, then, delete the `tracing-tower-http` crate in favour of similar functionality in `tracing-tower`?

**Comment by LucioFranco** [comment]
I am +1 on punting on this and moving it to `tower-rs/tower/tower-tracing`

**Comment by davidbarsky** [comment]
Will be brief, gotta run to a meeting, but:

> Should this crate [ed: `tracing-tower`] even live in tracing's github repo?

I think this should probably be moved into Tower's GitHub repo, but I'm iffy on that. I don't have a strong enough opinion.

> The tracing-tower-http crate provides batteries-included functionality for instrumenting services whose request and response types are http::Request/http::Response.

Merge, for sure.

**Comment by little-dude** [comment]
Are there plans to make a release of `tracing-tower`? Even a `0.0.1`?

**Comment by hawkw** [comment]
@little-dude yup, it's on my to-do list. I think the API complexity can be reduced pretty significantly, though. 

I'll see if I have the chance to give that a pass this afternoon.

**Comment by little-dude** [comment]
Thanks a lot @hawkw :) Do you need help with something here?

**Comment by hdevalence** [comment]
I'd be interested in using a published `tracing-tower` too; there's no rush but it would be convenient to avoid having a git dependency on `tracing`.

**Comment by hawkw** [comment]
> I'd be interested in using a published `tracing-tower` too; there's no rush but it would be convenient to avoid having a git dependency on `tracing`.

it would be nice to see which parts you're actually using, since i'd like to try and clean up the API surface a bit --- i think the `MakeService` impls may not be necessary, for example.

**Comment by hdevalence** [comment]
This was enough for me: https://github.com/penumbra-zone/tower-abci/blob/437f35d647862d68864ec6ad59fcba7dbc6d9d0d/src/split.rs#L30-L31
